org.OpenT2T.Sample.SuperPopular.Thermostat/Nest Thermostat/js/thingTranslator.js

```diff
@@ -0,0 +1,179 @@
+'use strict';
+
+var https = require('follow-redirects').https;
+var httpsNoRedirect = require('https');
+
+// logs device state
+function logDeviceState(device) {
+    if (typeof (device) !== 'undefined') {
+        console.log('  device.name          : ' + device.name);
+        console.log('  device.props         : ' + device.props);
+    } else {
+        console.log('device is undefined');
+    }
+};
+
+var deviceId, accessToken;
+
+// module exports, implementing the schema
+module.exports = {
+
+    device: null,
+
+    initDevice: function(dev) {
+        this.device = dev;
+
+        if (typeof this.device != 'undefined') {
+            if (typeof (this.device.props) !== 'undefined') {
+                var props = JSON.parse(this.device.props);
+
+                if (typeof (props.access_token) !== 'undefined') {
+                    accessToken = props.access_token;
+                } else {
+                    console.log('props.access_token is undefined.');
+                }
+
+                if (typeof (props.id) !== 'undefined') {
+                    deviceId = props.id;
+                } else {
+                    console.log('props.id is undefined.');
+                }
+            } else {
+                console.log('props is undefined.');
+            }
+        } else {
+            console.log('device is undefined.');
```

Comment:
Invalid arguments should result in thrown Errors instead of just logging. (I know you copy-pasted this part from other translators.) This can be made much cleaner using [a validation helper function like this](https://github.com/openT2T/translators/blob/master/org.OpenT2T.Sample.SuperPopular.Lamp/Philips%20Hue%20Light%20Bulb/js-promise/thingTranslator.js#L33).